### PR TITLE
fix: add change orientation event with modal render

### LINKED ISSUE #1132: Modal layout is incorrect after Orientation change
## 🐛 Bug Report

the Modal content position is not correct after screen orientation is changed to landscape and back to portrait.

## To Reproduce

Rotate the screen to landscape and then rotate back to portrait view. Open a modal, the modal is not centered.

## Expected behavior

The modal should be centered in window with the correct left and top.

## UI Kitten and Eva version

| Package      | Version |
| ----------- | ----------- |
| @eva-design/eva      |    5.0.0   |
| @ui-kitten/components   |    5.0.0     |

## Environment information

Xnip2020-06-07_00-19-24

### PR DESCRIPTION
**Pull request by whitestranger7**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:
The solutions is quite simple - I just render new modal with new id. I dont think there would be a possibility to apply new position to old one. If you think we can apply new position for the modal - let me know pls

Closes #1132

### CODE REVIEW

File: src/components/ui/modal/modal.component.tsx [hunk 1]
```diff
@@ -93,7 +94,15 @@ export class Modal extends React.PureComponent<ModalProps, State> {
     this.modalId = ModalService.hide(this.modalId);
   };
 
+  public forceRender = (): void => {
```
Comment (artyorsh) [inline src/components/ui/modal/modal.component.tsx:97]:
```suggestion
  private forceRender = (): void => {
```

Should not be public, should it?
If not, it should be declared below lifecycle methods

**Review by artyorsh — COMMENTED** [review]
`forceRender` should not be public most likely. I would suggest making it private and declaring it below the lifecycle methods then. If we leave it public, it's better to call that `forceUpdate` or simply `update`.

**Comment by malashkevich** [comment]
Will it work correctly if the modal will contain some state? For example, if the user will put there `Input` and then change the orientation.
Anyway I would say that we need recalculate the original modal rather than hide/show

**Comment by whitestranger7** [comment]
> Will it work correctly if the modal will contain some state? For example, if the user will put there `Input` and then change the orientation.
> Anyway I would say that we need recalculate the original modal rather than hide/show

Sure thing, it works fine with any children component inside. I've updated the method and made it private. Take a look

File: src/components/ui/modal/modal.component.tsx [hunk 2]
```diff
@@ -117,9 +119,16 @@ export class Modal extends React.PureComponent<ModalProps, State> {
   }
 
   public componentWillUnmount(): void {
+    Dimensions.removeEventListener('change', this.forceRender);
```
Comment (malashkevich) [inline src/components/ui/modal/modal.component.tsx:122]: As long as this is listener consider renaming it to "on%%NAME_OF_THE_EVENT%%"

**Review by artyorsh — APPROVED** [review]

**Review by malashkevich — APPROVED** [review]

**Pull request merged**